### ISSUE TITLE
Explicitly handle exceptions

### ISSUE THREAD

**Issue by sckott** [sckott/pytaxize#3]
Explicitly handle exceptions - for example, calling `gnr_resolve('abcdefg')` gives  `"KeyError: 'results'"` - give an exception in plain English what happened.

**Comment by akshayah3** [comment]
I was testing all the methods with `invalid` inputs and quite a few were giving a not so helpful error message. These methods need specific input validation for example `gbif_parse(scientificname)` raises an error when `scientificname` is not a list. I think we can validate these methods to provide a more helpful error message. 

I will try to tackle these methods one by one. Is that ok? Or should I work on this and send a single pull request?

**Comment by sckott** [comment]
Yes, please do!  Smaller PRs are better than one big PR

**Comment by akshayah3** [comment]
@sckott Sure. I will work on this.

**Comment by akshayah3** [comment]
@sckott I have checked for all the functions. I think this issue can be considered fixed.

**Comment by sckott** [comment]
great, thanks
